drivers/hmis/spec/requests/hmis/join_household_spec.rb

```diff
@@ -219,6 +219,10 @@ def perform_mutation(
           donor_child.reload
         end.to change(donor_hoh, :active_unit_occupancy).to(nil).
           and change(donor_child, :active_unit_occupancy).to(nil)
+
+        expect(donor_hoh.unit_occupancies.count).to eq(1) # Still has historical occupancy (regression #8379)
+        expect(donor_hoh.unit_occupancies.sole.occupancy_period).to be_present
+        expect(donor_hoh.unit_occupancies.sole.end_date).to be_present
       end
```

Comment:
It seems like both enrollments are meant to release their units (donor_hoh and child_hoh) so I'd suggest updating the test to test that. We can add the assertions to the existing "expect" block which I think is a little neater:

```suggestion
      let(:today) { Date.current }

      it 'removes the current unit from the joiners' do
        expect do
          perform_mutation
          [donor_hoh, donor_child, donor_hoh_occupancy, donor_child_occupancy].each(&:reload)
        end.to change(donor_hoh, :active_unit_occupancy).to(nil).
          and change(donor_child, :active_unit_occupancy).to(nil).
          # Unit is released from the joiners, and historical unit occupancy is still present (regression #8379)
          and change { donor_hoh_occupancy.occupancy_period.end_date }.from(nil).to(today).
          and change { donor_child_occupancy.occupancy_period.end_date }.from(nil).to(today)
      end
```

Reply:
https://github.com/greenriver/hmis-warehouse/pull/5968/commits/d82db7351daa1d069aaf41f97196b05ea2087cf5